aws/modules/infrastructure_modules/eks/variables.tf

```diff
@@ -139,3 +140,33 @@ variable "image_gc_low_threshold_percent" {
   type        = number
   default     = 80
 }
+
+variable "node_security_group_additional_rules" {
+  description = "List of additional security group rules to add to the node security group created. Set source_cluster_security_group = true inside rules to set the cluster_security_group as source"
+  type        = any
+  default = {
+    ingress_self_all = {
+      description = "Node to node all ports/protocols"
+      protocol    = "-1"
+      from_port   = 0
+      to_port     = 0
+      type        = "ingress"
+      self        = true
+    }
+    egress_all = {
+      description      = "Node all egress"
+      protocol         = "-1"
+      from_port        = 0
+      to_port          = 0
+      type             = "egress"
+      cidr_blocks      = ["0.0.0.0/0"]
+      ipv6_cidr_blocks = ["::/0"]
+    }
+  }
+}
+
+variable "node_security_group_enable_recommended_rules" {
+  description = "Determines whether to enable recommended security group rules for the node security group created. This includes node-to-node TCP ingress on ephemeral ports and allows all egress traffic"
+  type        = bool
+  default     = true
+}
```

Comment:
What's the difference between the rules above and these rules? Are they rules provided by the module?

Reply:
There are some rules in the recommendation that we dont want to use in our project, e.g egress on all ports from anywhere. So moving this to vairable so that we can set our own rules. Below are the recommended rules from original eks module which we are using
![Screenshot 2024-06-12 at 15 58 05](https://github.com/Ensono/stacks-terraform/assets/78567578/117fe4cd-a96b-46f6-853d-db3a263c20d5)

Reply:
we dont want 'Allow all egress' permission

Reply:
and somewhere else, it could be others, so just moving it to vars so that we are not tied to these default rules

Reply:
Could you please pop a description into the `README.md` about how to set this up for your own rules?